xray/httptrace.go

```diff
@@ -25,6 +25,12 @@ type HTTPSubsegments struct {
 	responseCtx context.Context
 }
 
+// NewHTTPSubsegments creates a new HTTPSubsegments to use in
+// httptrace.ClientTrace functions
+func NewHTTPSubsegments(opCtx context.Context) *HTTPSubsegments {
```

Comment:
Could you tell me more about this use case? It seems like it just uses a different way to pass opCtx params.

Reply:
Sure.

You can't assign an unexported field from outside a package.

Example:

```go
// $GOPATH/src/github.com/superfly/unexportedtest/somepkg/somepkg.go
package somepkg

type ExportedStruct struct {
	aField string
}

// $GOPATH/src/github.com/superfly/unexportedtest/main.go
package main

import "github.com/superfly/unexportedtest/somepkg"

func main() {
	some := somepkg.ExportedStruct{aField: "hello"}
}
```

Result:

```
$ go build .
# github.com/superfly/unexportedtest
./main.go:6:39: unknown field 'aField' in struct literal of type somepkg.ExportedStruct
```

I wanted to use `HTTPSubegments` from my own package but I couldn't initialize it with a context since the field `opCtx` is not exported (capitalized).

Creating this `NewHTTPSubsegments` function fixes that issue.

Reply:
Thank you for your example and now I understand your use case. Basically you want use `HTTPSubsegment` struct but every item under this struct is lowercase so that you cannot initialize. I will merge this fix and thank you again for your fix.